src/main/java/org/jenkinsci/plugins/stashNotifier/StashNotifier.java

```diff
@@ -636,21 +672,50 @@ private HttpEntity newStashBuildNotificationEntity(
         json.put("description", getBuildDescription(build, state));
         json.put("url", Jenkins.getInstance()
         		.getRootUrl().concat(build.getUrl()));
-        
+
         return new StringEntity(json.toString(), "UTF-8");
 	}
 
+
+
 	/**
-	 * Returns the build key used in the Stash notification. Includes the 
+	 * Returns the build key used in the Stash notification. Includes the
 	 * build number depending on the user setting.
-	 * 
+	 *
 	 * @param 	build	the build to notify Stash of
 	 * @return	the build key for the Stash notification
 	 */
-	private String getBuildKey(final AbstractBuild<?, ?> build) {
-		StringBuilder key = new StringBuilder();
+	private String getBuildKey(final AbstractBuild<?, ?> build,
+							   BuildListener listener) {
+
+        String overriddenKey = (projectKey != null && projectKey.trim().length() > 0) ? projectKey : getDescriptor().getProjectKey();
+
+		if (overriddenKey != null && overriddenKey.trim().length() > 0) {
+			PrintStream logger = listener.getLogger();
+			try {
+				EnvVars environment = build.getEnvironment(listener);
+				return StringEscapeUtils.escapeJavaScript(environment.expand(projectKey));
```

Comment:
By returning from here, you effectively prevent the evaluation of  `prependParentProjectKey` and `includeBuildNumberInKey` settings. While I think this makes sense, I propose to add a corresponding remark to the tooltip of the field _Override project key_ to avoid confusion.